misc/pytorch_toolkit/breast_ultrasound_simulation/src/model.py

```diff
@@ -0,0 +1,419 @@
+        self.xdim = nn.Conv2d(
+            ch, ch, (1, k), padding=(
+                0, p), stride=(
+                1, s), groups=ch, bias=False)
+        self.bias = nn.Conv2d(
+            ch, ch, (1, 1), padding=(
+                0, 0), stride=(
+                1, 1), groups=ch, bias=True)
+
+    def forward(self, x):
+        # do y x first follwed by the other spectal decompsed filters!
+
+        op = self.ydim(x)
+        #print(op.shape, x.shape)
+        op = self.xdim(op)
+        op = self.bias(op)
+
+        return op
+
+
+class SpectralConv2dInter(nn.Module):
+    def __init__(self, ch, k, pad, dil, wts_list=None, a=1):
+        super().__init__()
+        self.kernel = k + (k - 1) * (dil - 1)  # default dil = 1(no dilation)
+        self.pad = pad
+        index = torch.Tensor([i for i in range(0, self.kernel, dil)]).to(torch.int64)
+        z = torch.zeros(ch, ch, self.kernel, 1)
+        if wts_list is None:
+            print("Not implemented")
+        ydim_wt = wts_list.ydim.weight.data
+
+        self.wts_norm = F.pad(
+            ydim_wt, (0, 0, (self.kernel - k) // 2, (self.kernel - k) // 2))
+        self.wts_dil = z.index_add_(2, index, ydim_wt)
+
+        self.wts = torch.nn.Parameter(
+            self.wts_norm * (1 - a) + self.wts_dil * a)
+
+        self.xdim = nn.Conv2d(
+            ch, ch, (1, k), padding=(
+                0, 1), stride=(
+                1, 1), groups=ch, bias=False)
+        self.bias = nn.Conv2d(
+            ch, ch, (1, 1), padding=(
+                0, 0), stride=(
+                1, 1), groups=ch, bias=True)
+
+        if wts_list is not None:
+            self.xdim.load_state_dict(wts_list.xdim.state_dict())
+            self.bias.load_state_dict(wts_list.bias.state_dict())
+
+    def interpolate(self, a):
+        self.wts.data = self.wts_norm * a + self.wts_dil * (1 - a)
+
+    def forward(self, x):
+
+        op = F.conv2d(x, self.wts, padding=(self.pad, 0))
+
+        op = self.xdim(op)
+        op = self.bias(op)
+        return op
+
+
+class SpectralConv3dInter(nn.Module):
+    def __init__(self, ch, k, pad, dil):
+        super().__init__()
+        self.kernel = k + (k - 1) * (dil - 1)  # default dil = 1(no dilation)
+        self.pad = pad
+        p = int((pad == 2))
+        self.ydim = nn.Conv3d(
+            ch, ch, (1, self.kernel, 1), padding=(
+                0, pad, 0), stride=(
+                1, 1, 1), bias=False)
+        self.xdim = nn.Conv3d(
+            ch, ch, (1, 1, k), padding=(
+                0, 0, p), stride=(
+                1, 1, 1), groups=ch, bias=False)
+        self.zdim = nn.Conv3d(
+            ch, ch, (k, 1, 1), padding=(
+                p, 0, 0), stride=(
+                1, 1, 1), groups=ch, bias=False)
+        self.bias = nn.Conv3d(
+            ch, ch, (1, 1, 1), padding=(
+                0, 0, 0), stride=(
+                1, 1, 1), groups=ch, bias=True)
+
+        """if(wts_list is not None):
+            print("In asssss")
```

Comment:
What should this mean?

Reply:
Sorry. This was to be deleted before PR.